File: Project/Javascript/Number-guessing/demoNG.js
const readLine = require("readline");

const rl = readLine.createInterface({
  input: process.stdin,
  output: process.stdout,
});

console.log("Hi Welcome");
console.log("U Have 7 chances");
console.log("Lets Start :- ");

let low, high;

let nums;
let chance = 7;
let guessCount = 0;

rl.question("Entre The lower Bound : - ", (lowInput) => {
  low = parseInt(lowInput);

  rl.question("Entre The Higher Bound :- ", (highInput) => {
    high = parseInt(highInput);

    nums = Math.floor(Math.random() * (high - low + 1)) + low;
    console.log("\n u hv 7 chance to guess");
    console.log("Lets Start");
    askGuess();
  });
});

function askGuess() {
  if (guessCount >= chance) {
    console.log(`Sorry the nums was ${nums} try next`);
  }
  rl.question("Entre You Guess :- ", (guesInput) => {
    let guess = parseInt(guesInput);

    guessCount++;
    if (guess === nums) {
      console.log(`Correct ! the number ${nums} you guessed`);
      rl.close();
    } else if (guess > nums) {
      console.log("Too High ! try lower");
      askGuess();
    } else if (guess < nums) {
      console.log("Too Low ! try Higher");
      askGuess();
    }
  });
}

File: Project/Python/Number-guessing/demoNG.py
import random

print("Hi! Wlcome to the Number guessing Game. \nYou have 7 chance To guess.......... \nLet's Start :-")

low = int(input("Entre The Lower Bound :- "))
high = int(input("Entre The Higher Bound :- "))

print(f"\nYou & chances To guess The NUmber Between {low} and {high} . \nLet's Start....! ")

num = random.randint(low,high)
ch = 7 
gc  = 0

while gc < ch :
    gc +=1
    guess = int(input("Entre Your Guess :- "))

    if guess == num :
        print(f'Correct ! the Number is {num}. you guessed it in the {gc} attempts.')
        break
    elif gc>=ch and guess !=num:
        print(f'Sorry! The number was {num}. Better luck next time.')
    elif guess > num:
        print("Too High! Try a lower number.")
    elif guess < num:
        print("Too Low! Try a higher number.")

    if gc == ch:
         print(f"Sorry! The number was {num}. Better luck next time.")